### PR TITLE
Add xMSannotator

### PR DESCRIPTION
**Pull request by maximskorik**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/r-xmsannotator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - r-purrr
+    - r-tibble
+    - r-data.table
+    - bioconductor-rhdf5
+    - r-rcpp
+    - r-arrow
+
+  test:
+    commands:
+      - $Rscript -e "library('xmsannotator')"
```
Comment (fanninpm) [inline recipes/r-xmsannotator/meta.yaml:45]: This block should be at root level.

```suggestion
test:
  commands:
    - $Rscript -e "library('xmsannotator')"
```

Comment (maximskorik) [reply recipes/r-xmsannotator/meta.yaml]: Thank you, I couldn't wrap my head around what was the problem.

**Comment by fanninpm** [comment]
You can just simply add a license statement inside the `about:` section (essentially reverting 5bfc5c9) to satisfy the lint check. It'll let you off with a warning if you do that. (I personally recommend going for the [MIT License](https://choosealicense.com/licenses/mit/), but it's ultimately your choice.)

**Comment by maximskorik** [comment]
`test-linux` was failing before I changed version from "1.3.2+recetox0" to "1.3.2", is this expected behaviour or the problem may have been somewhere else? In [Conda Docs](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#version-ordering) it's stated that one can use "+" in version.

**Comment by fanninpm** [comment]
I tried [these troubleshooting steps](https://bioconda.github.io/contributor/troubleshooting.html#troubleshooting-failed-mulled-build-tests) with a different recipe, but I essentially got nowhere. Failing that, I would ping the core team.

On a completely different note, I was curious why you weren't using `https://github.com/RECETOX/xMSannotator/archive/refs/tags/{{ 
version }}.tar.gz` as your source URL.

**Review by dpryan79 — APPROVED** [review]

**Comment by maximskorik** [comment]
> I tried [these troubleshooting steps](https://bioconda.github.io/contributor/troubleshooting.html#troubleshooting-failed-mulled-build-tests) with a different recipe, but I essentially got nowhere. Failing that, I would ping the core team.
> 
> On a completely different note, I was curious why you weren't using `https://github.com/RECETOX/xMSannotator/archive/refs/tags/{{ version }}.tar.gz` as your source URL.

Thank you for assistance :)
It's that we don't have a fully Conda-compatible version of the tool yet hence I was testing it with "packaging" branch. Once we release the new version I will change the URL to the one that you mentioned.

File: recipes/r-xmsannotator/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - r-rcpp
+    - r-arrow
+
+test:
+  commands:
+    - $R -e "library('xmsannotator')"
+
+about:
+  home: "{{ github }}"
+  license: MIT
```
Comment (martenson) [inline recipes/r-xmsannotator/meta.yaml:49]: The code is licensed under GPL2.0. See https://github.com/yufree/xMSannotator/blob/master/DESCRIPTION#L10

File: recipes/r-xmsannotator/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  summary: Annotate peak intensity table with compounds from the compounds database.
+
+extra:
+  maintainers:
```
Comment (martenson) [inline recipes/r-xmsannotator/meta.yaml:53]: I'd recommend putting `RECETOX` here for ensuring continuity.

**Comment by martenson** [comment]
@BiocondaBot update

**Comment by martenson** [comment]
@BiocondaBot please update

**Comment by fanninpm** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**